### DISCUSSION TITLE
Create new Resource X modal from the table of Resource Y

### DISCUSSION THREAD

**Discussion by Sanabria** [filamentphp/filament#4846] [Help]
I just started using Filament and loving it. However I ran into a bump:

I have two resources `booths` and `reports`. When we are in the list page of `booths` and we click in one of the rows of the table, I would like to fire a modal with a form to create a new `report` associated to the clicked `booth`.

I've tried reading the docs but haven't been able to piece together what would be the best approach to accomplish this. I'd really appreciate if somebody could point me in the right direction. Thanks in advance.

**Comment by danharrin** [comment]
Try something like this:

```php
Action::make('createReport')
    ->form([
        // form components
    ])
    ->action(function (Booth $record, array $data) {
        $record->reports()->create($data);

        Notification::make()->title('Report created')->send();
    })
```

https://filamentphp.com/docs/2.x/tables/actions

**Reply by Sanabria** [reply]
@danharrin Thank you so much! I was indeed able to make it work using your suggestion!

Any ideas on how can I create an `edit` form in a similar fashion?

**Reply by danharrin** [reply]
Editing a booth?
